File: 75-sort-colors/75-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int l = 0;
        int p = 0;
        int r = nums.size() - 1;
        
        while (p <= r) {
            
            switch (nums[p]) {
            
                case 0:
                    swap(nums[l++], nums[p++]);
                    break;
                    
                case 1:
                    p++;
                    break;
                    
                case 2:
                    swap(nums[r--], nums[p]);
            }
        }
    }
};

File: 75-sort-colors/75-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let p = 0;
    let l = 0;
    let r = nums.length - 1;
    
    while (p <= r) {
        
        if ( nums[p] === 2) {
            nums[p] = nums[r];
            nums[r] = 2;
            r -= 1;
        }
        else if ( nums[p] === 0 ) {
            nums[p] = nums[l];
            nums[l] = 0;
            l += 1;
            p += 1;
        }
        else {
            p += 1;
        }
    }
};